File: Chapter7/RandomGenerator.py

import sys, random

def rng04():
    return random.randint(0,4);

def rng03():
    while True:
        i = rng04()
        if i<4:
            return i

def rng01():
    while True:
        i = rng04()
        if i<4:
            return i%2

def rng07():
    i = rng03()
    j = rng01()
    return i*2+j

if __name__ == "__main__":
    for i in range(10000):
        print( rng07() )
    sys.exit()

File: Chapter7/RandomGenerator.java
package Chapter7.RandomGenerator;

import java.util.Random;

public class RandomGenerator
{

    private int rng04()
    {
        return new Random().nextInt(5);
    }

    private int rng03()
    {
        while(true)
        {
            int n = rng04();
            if(n<4)
                return n;
        }
    }

    private int rng01()
    {
        while(true)
        {
            int n = rng04();
            if(n<4)
                return n%2;
        }
    }

    public int rng07()
    {
        int i = rng03();
        int j = rng01();
        return i*2+j;
    }

    public static void main(String[] args)
    {
        RandomGenerator r = new RandomGenerator();
        for(int i=0; i<10000; i++)
            System.out.println(r.rng07());
    }
}
